engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/StreamProcessor.java

```diff
@@ -264,7 +264,8 @@ private long recoverFromSnapshot() {
         zeebeState.getLastProcessedPositionState().getLastSuccessfulProcessedRecordPosition();
 
     final boolean failedToRecoverReader = !logStreamReader.seekToNextEvent(snapshotPosition);
-    if (failedToRecoverReader) {
+    if (failedToRecoverReader
+        && processingContext.getProcessorMode() == StreamProcessorMode.PROCESSING) {
```

Comment:
:thought_balloon: I'm not sure if we really need this check, even for processing. I guess that Raft ensures that the right node will be the next leader.

Reply:
Raft ensures that this node has the latest data. So it is expected that the it will never fail here. But could there by any issues that can cause this to fail - such as bugs, data corruption etc? The question is should we guard against those?

Reply:
We can leave the check for now. I just had the feeling it is not necessary :sweat_smile: